tree/dataframe/src/RDFInterfaceUtils.cxx

```diff
@@ -340,16 +340,15 @@ std::vector<std::string> FindUsedColumnNames(std::string_view expression, const
    // To help matching the regex
    const std::string paddedExpr = " " + std::string(expression) + " ";
    static const std::string regexBit("[^a-zA-Z0-9_]");
-   Ssiz_t matchedLen;
 
-   std::vector<std::string> usedBranches;
+   std::set<std::string> usedBranches;
 
    // Check which custom columns match
    for (auto &brName : customColumns) {
       std::string bNameRegexContent = regexBit + brName + regexBit;
-      TRegexp bNameRegex(bNameRegexContent.c_str());
-      if (-1 != bNameRegex.Index(paddedExpr.c_str(), &matchedLen)) {
-         usedBranches.emplace_back(brName);
+      TPRegexp bNameRegex(bNameRegexContent.c_str());
+      if (0 != bNameRegex.Match(paddedExpr.c_str())) {
```

Comment:
Passing nMaxMatches=1 to Match might be slightly faster thanks to early returns

Reply:
this is actually a very nice suggestion which I will try out immediately. Thanks for spotting this!